OptimoveSDK/Sources/Classes/Optimobile/InApp/InAppPresenter.swift

```diff
@@ -130,38 +160,31 @@ class InAppPresenter : NSObject, WKScriptMessageHandler, WKNavigationDelegate{
         }
         
         messageQueueLock.wait()
-        defer {
-            messageQueueLock.signal()
-        }
         
         messageQueue.removeObject(at: 0)
         pendingTickleIds.remove(message.id)
         currentMessage = nil
         
         if messageQueue.count == 0 {
             pendingTickleIds.removeAllObjects()
-            self.destroyViews()
```

Comment:
View cleanup is now included in presentFromQueue().

This allows simplifying the conditional nesting below, and it is more consistent with the Android SDK implementation so it is easier to compare the changes for equivalence.